chacra/tests/models/test_repos.py

```diff
@@ -38,3 +38,33 @@ def test_created_no_binaries_is_not_generic(self, session):
             distro_version='7',
             )
         assert repo.is_generic is False
+
+
+class TestInferType(object):
```

Comment:
Do we need a test to ensure that a 'related' repo would get it's type set correctly if a binary is uploaded for a project that the 'related' repo includes?